src/freenet/clients/http/DiagnosticToadlet.java

```diff
@@ -441,15 +441,13 @@ private StringBuilder threadsStats() {
 			.getThreadDiagnostics();
 
 		List<NodeThreadInfo> threads = threadDiagnostics.getThreads();
-
-		ThreadMXBean threadMxBean = ManagementFactory.getThreadMXBean();
-		double totalCpuTime = Arrays.stream(threadMxBean.dumpAllThreads(false, false))
-			.mapToDouble(
-				(thread) -> threadMxBean.getThreadCpuTime(thread.getThreadId())
-			).sum();
-
 		threads.sort(Comparator.comparing(NodeThreadInfo::getCpuTime).reversed());
 
+		double totalCpuTime = threads
```

Comment:
You know what I just realized as I wondered why you would do this again? At this point we don’t actually know how much time was spent between two measurements used to create the delta. And as I don’t want this toadlet to fiddle around with the configuration I think we should put the list of thread infos _and_ the timing information into a new object… maybe a `ThreadDeltaSnapshot`? Hmm, sounds horrible. What do you think?

Reply:
I considered this myself but I wanted to keep it as simple as possible. ThreadsSnapshot is not too bad with ThreadsSnapshot#getDelta, ThreadsSnapshot#getThreads and it can grow to contain more metadata.

Reply:
Heh, yes, there’s “simple” and there’s “simple.” 😁 

As long as we thought we’d only need the list of threads the list of threads was enough but now that we see that we need more it’s totally fine to add a new class into this that contains exactly what we need.

Reply:
Done, created NodeThreadSnapshot class to support threads list, total cpu time and interval. Interval for now is the configured for the Runner, could be calculated at each executing for further precision but I would argue such precision is not required now.